### PR TITLE
Avoid fixing permissions; add regexp alias file, add AWS SES outgoing email support

### PR DESCRIPTION
**Pull request by tve**
This PR is not intended to be merged as-is, please indicate whether you want all/some/none of it and I can cherry-pick accordingly. It has:
- avoid fixing permissions if they're OK #190
- support regexp alias file
- support outgoing email via AWS SES

I added docs to the README, but I would move that to the wiki.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,32 @@
-# docker-mailserver [![Build Status](https://travis-ci.org/tomav/docker-mailserver.svg?branch=master)](https://travis-ci.org/tomav/docker-mailserver)
+# docker-mailserver [![Build Status](https://travis-ci.org/tve/docker-mailserver.svg?branch=master)](https://travis-ci.org/tve/docker-mailserver)
+
+This is a fork of https://github.com/tomav/docker-mailserver with some additional features
+described below.
+
+### Sending outbound mail via Amazon SES
```
Comment (tomav) [inline README.md]: Could you move this doc to the Wiki? Simpler to maintain.

File: README.md [hunk 2]
```diff
@@ -1,4 +1,32 @@
-# docker-mailserver [![Build Status](https://travis-ci.org/tomav/docker-mailserver.svg?branch=master)](https://travis-ci.org/tomav/docker-mailserver)
+# docker-mailserver [![Build Status](https://travis-ci.org/tve/docker-mailserver.svg?branch=master)](https://travis-ci.org/tve/docker-mailserver)
```
Comment (tomav) [inline README.md]: This part should not be commit to this repo.

File: target/start-mailserver.sh [hunk 3]
```diff
@@ -78,6 +78,15 @@ if [ -f /tmp/docker-mailserver/postfix-virtual.cf ]; then
 else
   echo "==> Warning: 'config/postfix-virtual.cf' is not provided. No mail alias/forward created."
 fi
+if [ -f /tmp/docker-mailserver/postfix-regexp.cf ]; then
+  # Copying regexp alias file
```
Comment (tomav) [inline target/start-mailserver.sh]: Could you add test on it (see `test/bats.tests`)

File: target/start-mailserver.sh [hunk 4]
```diff
@@ -216,8 +246,13 @@ else
   echo "==> Warning: 'SASL_PASSWD' is not provided. /etc/postfix/sasl_passwd not created."
 fi
 
-echo "Fixing permissions"
-chown -R 5000:5000 /var/mail
+# Fix permissions, but skip this if 3 levels deep the user id is already set
```
Comment (tomav) [inline target/start-mailserver.sh]: There's a pending PR from @Josef-Friedrich https://github.com/tomav/docker-mailserver/pull/193
You should discuss the implementation to only keep one.
Thanks.

File: Dockerfile [hunk 5]
```diff
@@ -5,7 +5,7 @@ MAINTAINER Thomas VIAL
 RUN DEBIAN_FRONTEND=noninteractive apt-get update -q --fix-missing && \
 	apt-get -y upgrade && \
 	apt-get -y install --no-install-recommends \
-	postfix dovecot-core dovecot-imapd dovecot-pop3d dovecot-sieve dovecot-managesieved gamin amavisd-new spamassassin razor pyzor \
+	postfix dovecot-core dovecot-imapd dovecot-pop3d dovecot-sieve dovecot-managesieved gamin amavisd-new spamassassin razor pyzor libsasl2-modules \
```
Comment (tomav) [inline Dockerfile]: `libsasl2-modules` is really needed?

File: README.md (hunk 1 above)
Comment (tomav) [inline README.md]: https://github.com/tomav/docker-mailserver/wiki

**Comment by tomav** [comment]
Thanks for the work @tve 
Let's check these little things and I'll merge.

File: README.md (hunk 1 above)
Comment (tve) [inline README.md]: That was the plan :-)

File: README.md (hunk 2 above)
Comment (tve) [inline README.md]: Yup

File: Dockerfile (hunk 5 above)
Comment (tve) [inline Dockerfile]: Unfortunately, yes. The config you have uses dovecot for SASL, but dovecot doesn't support client auth (http://www.postfix.org/SASL_README.html#client_sasl says "At this time, the Dovecot SASL implementation does not provide client functionality") and what is needed for AWS SES is for postfix to do SASL as a client. The same would be needed for pretty much any scenario where outgoing email gets relayed through an authenticated connection, unless I'm missing something...

File: target/start-mailserver.sh (hunk 3 above)
Comment (tve) [inline target/start-mailserver.sh]: Done

File: target/start-mailserver.sh (hunk 4 above)
Comment (tve) [inline target/start-mailserver.sh]: Started...

**Comment by tve** [comment]
This PR is ready from my point of view. Please let me know if there's anything else you would like changed.

**Comment by tomav** [comment]
Thank you @tve, really nice work and wiki documentation.

**Pull request merged**